### ISSUE TITLE
gcc 10.2/glibc2.17 for arm won't build unless kernel shared modules enabled

### ISSUE THREAD

**Issue by artiommorozov** [crosstool-ng/crosstool-ng#1449]
Glibc final stage builds iconv which requires libgcc_eh.

```
[ALL  ]      aarch64-unknown-linux-gnu-gcc  -g -O2 -U_FORTIFY_SOURCE   -march=armv8-a         -nostdlib -nostartfiles -static -o /home/artiom/cross/try1/.build/aarch64-unknown-linux-gnu/build/build-libc-final/multilib/iconv/iconvconfig    /home/artiom/cross/try1/.build/aarch64-unknown-linux-gnu/build/build-libc-final/multilib/csu/crt1.o /home/artiom/cross/try1/.build/aarch64-unknown-linux-gnu/build/build-libc-final/multilib/csu/crti.o `aarch64-unknown-linux-gnu-gcc  -g -O2 -U_FORTIFY_SOURCE   -march=armv8-a          --print-file-name=crtbeginT.o` /home/artiom/cross/try1/.build/aarch64-unknown-linux-gnu/build/build-libc-final/multilib/iconv/iconvconfig.o /home/artiom/cross/try1/.build/aarch64-unknown-linux-gnu/build/build-libc-final/multilib/iconv/strtab.o /home/artiom/cross/try1/.build/aarch64-unknown-linux-gnu/build/build-libc-final/multilib/iconv/xmalloc.o /home/artiom/cross/try1/.build/aarch64-unknown-linux-gnu/build/build-libc-final/multilib/iconv/hash-string.o  -Wl,--start-group /home/artiom/cross/try1/.build/aarch64-unknown-linux-gnu/build/build-libc-final/multilib/libc.a -lgcc -lgcc_eh  -Wl,--end-group `aarch64-unknown-linux-gnu-gcc  -g -O2 -U_FORTIFY_SOURCE   -march=armv8-a          --print-file-name=crtend.o` /home/artiom/cross/try1/.build/aarch64-unknown-linux-gnu/build/build-libc-final/multilib/csu/crtn.o
[ALL  ]      /home/artiom/cross/try1/.build/aarch64-unknown-linux-gnu/buildtools/lib/gcc/aarch64-unknown-linux-gnu/10.2.0/../../../../aarch64-unknown-linux-gnu/bin/ld: cannot find -lgcc_eh
```
However libgcc_eh is not built by gcc with --disable-shared option, from gcc's libgcc/Makefile.in

```
all: libgcc.a
<skip>
ifeq ($(enable_shared),yes)
all: libgcc_eh.a libgcc_s$(SHLIB_EXT)
```

 which comes from scripts/build/cc/gcc.sh  stacktrace
do_gcc_core_backend("mode=static") 
do_cc_core_pass_2

because of 
```
case "${CT_THREADS}" in
        nptl)
            if [ "${CT_SHARED_LIBS}" = "y" ]; then
                core_opts+=( "mode=shared" )
            else
                core_opts+=( "mode=static" )
```

I'm not really sure how to fix it correctly: make a patch to glibc to make libgcc_eh optional? Patch gcc.sh to for make libgcc_eh.a here
[DEBUG]    ==> Executing:  'make' '-j5' '-l' 'all-gcc' 'all-target-libgcc'
? Or is that not supposed to be built that way at all? Please advise.

version is  1.24.0.254_fcf3233
build log attached
[build.log](https://github.com/crosstool-ng/crosstool-ng/files/5834373/build.log)

**Comment by cpackham** [comment]
Have you tried a newer glibc version? 2.17 is quite old.

**Comment by artiommorozov** [comment]
No, i haven't. I will try to check it out later next week if newer glibc still requires libgcc_eh.

**Comment by artiommorozov** [comment]
Tried newer glibc 2.26 kernel 4.1, building it with --disable-shared fails while installing start files due to
```
[EXTRA]      Installing C library start files
[ERROR]      cc1: fatal error: /home/artiom/cross/aarch64-glibc2.17/.build/aarch64-unknown-linux-gnu/build/build-libc-startfiles/multilib/libc-modules.h: No such file or directory
```

Same results on glibc 2.32 / linux 5.10. 

It looks like there's nothing to fix specifically for glibc 2.17. It was just a bit of surprise that shared libs setting from OS section in menuconfig affects glibc build :)

I think this issue can be closed now.
